crmsh/ui_configure.py

```diff
@@ -592,24 +592,47 @@ def do_filter(self, context, filterprog, *args):
     @command.skill_level('administrator')
     @command.completers(_id_list)
     def do_set(self, context, path, value):
-        "usage: set <path> <value>"
-        def split_path():
-            for oid in cib_factory.id_list():
-                if path.startswith(oid + "."):
-                    return oid, path[len(oid)+1:]
-            context.fatal_error("Invalid path: " + path)
-        obj_id, obj_attr = split_path()
+        """
+        usage: set <path> <value>
+
+        path:: id.[op_type.][interval.]name
+        """
+        path_errmsg = "Invalid path: \"{}\"; Valid path: \"id.[op_type.][interval.]name\"".format(path)
+        path_list = path.split('.')
+        if len(path_list) < 2 or len(path_list) > 4:
+            context.fatal_error(path_errmsg)
+
+        obj_id, *other_path_list = path_list
         rsc = cib_factory.find_object(obj_id)
         if not rsc:
-            context.fatal_error("Resource %s not found" % (obj_id))
-        nvpairs = rsc.node.xpath(".//nvpair[@name='%s']" % (obj_attr))
-        if not nvpairs:
-            context.fatal_error("Attribute not found: %s" % (path))
-        if len(nvpairs) != 1:
-            context.fatal_error("Expected 1 attribute named %s, found %s" %
-                                (obj_attr, len(nvpairs)))
+            context.fatal_error("Object {} not found".format(obj_id))
+
+        if len(other_path_list) == 1:
```

Comment:
Could we put some comment in top of the if to show which is the use case. Like `# id.name value use case`

Reply:
Changed, thanks!